gateway-js/src/__tests__/executeQueryPlan.test.ts

```diff
@@ -1114,13 +1068,11 @@ describe('executeQueryPlan', () => {
       const response = await executePlan(queryPlan, operation, undefined, schema);
 
       expect(response.data?.vehicle).toEqual(null);
-      expect(response.errors).toBeUndefined();
-      // SEE COMMENT ABOVE THIS TEST.  SHOULD BE RE-ENABLED AFTER #981 IS FIXED!
-      // expect(response.errors).toMatchInlineSnapshot(`
-      //   Array [
-      //     [GraphQLError: Abstract type "Vehicle" was resolve to a type [inaccessible type] that does not exist inside schema.],
-      //   ]
-      // `);
+      expect(response.errors).toMatchInlineSnapshot(`
```

Comment:
Note to reviewers: there is a bunch of comments on the background around this in `executeQueryPlan`, which is part about `postProcessingErrors`. The tl;dr is that the existing code was unconditionally ignoring post-processing errors, which some users have been complaining about in the past because it can make it very hard to debug issue with broken subgraphs, and an earlier patch tried to change this by unconditionally returning post-processing errors, but this was quickly reverted because post-processing errors often duplicate subgraphs errors and people were annoyed.

This patch implements a middle ground where we do not include post-processing errors at all if we have some query plan execution errors (thus make sure we can't create duplicates), but include those post-processing errors if there is no query plan execution errors whatsoever, to at least help in that case.

This typically make this test kind of work again: the test have a subgraph that returns an object for an `@inacessible` type, and the old was nullifying that object (correctly, we shouldn't return it) but was not indicating any error (which isn't great). The updated code bring back the error.

Do not that the new error message send differs from what graphQL-js would have produced, but I think the new message is a bit more helpful at pinpointing the problem.

Lastly, this code made me noticed that we still had some `cleanErrorOfInaccessible` code that was completely unused, so this patch removes it. Now that we control the code generating the errors, we have no reason to ever need to "clean" those error message: we can just make sure we don't leak anything in the first place (which is what is done here).